index.d.ts

```diff
@@ -33,7 +33,37 @@ declare namespace Fig {
   type InstallationScriptCompiler<T> = (_: { ctx: DotfileCompilationContext }) => T;
   type InstallationScript<T> = T | InstallationScriptCompiler<T>;
 
+  interface OnInstall {
+    /** The command to run to install */
+    command: InstallationScript<string>;
+    /** The check to run to verify the installation */
+    check: {
+      /** Check that commands exists */
+      commandExists?: InstallationScript<string[]>;
+      /** Check that files/folders exists */
+      fileExists?: InstallationScript<string[]>;
+      /** Check a commands output */
+      commandOutput?: {
+        /** The command to run to check the output */
+        command: InstallationScript<string>;
+        /** The expected output */
+        output: InstallationScript<string>;
```

Comment:
Not sure I understand the API here. Are these scripts that return a boolean value? Or are they returning a list of files, executables that is validated by the CLI? (I assume the later?)

Reply:
The later